File: crates/entities/src/entry.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Entry {
    pub post_slug: String,
    pub title: String,
    pub body: String,
    pub likes: Option<i64>,

    pub blog_slug: Option<String>,
    pub blog_name: Option<String>,

    pub author_slug: Option<String>,
    pub author_name: Option<String>,

    pub created_at: Option<String>,
    pub updated_at: Option<String>,
}

File: frontend/src/lib/types/Entry.ts
export default interface Entry {
  id: string
  slug: string

  title: string
  body: string
  likeCount: number

  blogSlug: string
  blogName: string

  authorSlug: string
  authorName: string

  createdAt: string
  updatedAt: string

  liked: boolean
  bookmarked: boolean
}
